functions/v2/test/index.test.js

```diff
@@ -13,94 +13,46 @@
 // limitations under the License.
 
 const assert = require('assert');
-const {request} = require('gaxios');
-const {exec} = require('child_process');
-const waitPort = require('wait-port');
+const {getTestServer} = require('@google-cloud/functions-framework/testing');
+const sinon = require('sinon');
+const supertest = require('supertest');
```

Comment:
I believe we moved away from `supertest` in the past, but don't quite remember why.

(Let's hold this thought until we figure out whether we want to use the FF as part of the testing architecture.)

cc @JustinBeckwith

Reply:
supertest seems to be used [throughout this repo](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/search?q=supertest). 

I am happy to revert the unit testing changes (all existing tests passed as written), although I strongly relying on this mechanism rather than forking processes for the reasons mentioned above (faster, less flaky).

Reply:
Ah - I didn't realize the _FF itself_ was providing a test server.

As long as we test the _integration_ with the FFs themselves, I'm fine with this.

We should file a follow-up issue to apply this change in other locations, too (assuming it's supported for Gen 1's "background" functions; if not, we should make sure to do it as part of the migration to Gen 2).

Reply:
These helpers only work with the new declarative APIs right now. We might be able to find a way to make them work with background functions though. I will file an issue in the Functions Framework repo.

Reply:
Filed issue here: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/issues/404